radixdlt-java/src/main/java/org/radix/utils/UInt256s.java

```diff
@@ -0,0 +1,95 @@
+package org.radix.utils;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * Utility methods for converting to/from {@link UInt256}.
+ *
+ * @author msandiford
+ */
+public final class UInt256s {
+	private UInt256s() {
+		throw new IllegalStateException("Can't construct");
+	}
+
+	private static final BigInteger MAX_UINT256 = toBigInteger(UInt256.MAX_VALUE);
+
+	/**
+	 * Returns the specified {@link UInt256} as a {@link BigInteger}.
+	 *
+	 * @param value The value to convert
+	 * @return The value as a {@link BigInteger}
+	 */
+	public static BigInteger toBigInteger(UInt256 value) {
+		// unsigned -> zero extend
+		byte[] bytes = new byte[UInt256.BYTES + 1];
+		bytes[0] = 0;
+		value.toByteArray(bytes, 1);
+		return new BigInteger(bytes);
+	}
+
+	/**
+	 * Returns the specified {@link UInt256} as a {@link BigDecimal}.
+	 *
+	 * @param value The value to convert
+	 * @return The value as a {@link BigDecimal}
+	 */
+	public static BigDecimal toBigDecimal(UInt256 value) {
+		return new BigDecimal(toBigInteger(value));
+	}
+
+	/**
+	 * Returns the specified {@link BigInteger} as a {@link UInt256}.
+	 *
+	 * @param value The value to convert
+	 * @return The value as a {@link UInt256}
+	 */
+	public static UInt256 fromBigInteger(BigInteger value) {
+		if (value.signum() < 0) {
+			throw new IllegalArgumentException("value must be >= 0: " + value);
+		}
+		if (value.compareTo(MAX_UINT256) > 0) {
+			throw new IllegalArgumentException("value must be <= " + MAX_UINT256 + ": " + value);
+		}
+		return UInt256.from(value.toByteArray());
+	}
+
+	/**
+	 * Returns the specified {@link BigInteger} as a {@link UInt256}.
+	 *
+	 * @param value The value to convert
```

Comment:
Would be helpful to know that this throws an `ArithmeticException`

Reply:
Javadoc added.